astropy/utils/masked/function_helpers.py

```diff
@@ -1552,30 +1552,62 @@ def _copy_of_mask(a):
 
 
 if NUMPY_LT_2_4:
-
     @dispatched_function
     def in1d(ar1, ar2, assume_unique=False, invert=False, *, kind=None):
         mask = _copy_of_mask(ar1).ravel()
         return _in1d(ar1, ar2, assume_unique, invert, kind=kind), mask, None
+else:
+    @dispatched_function
+    def in1d(ar1, ar2, /, assume_unique=False, invert=False, *, kind=None):
+        mask = _copy_of_mask(ar1).ravel()
+        return _in1d(ar1, ar2, assume_unique, invert, kind=kind), mask, None
```

Comment:
[`in1d` was removed in numpy 2.4](https://numpy.org/doc/stable/release/2.4.0-notes.html#removed-numpy-in1d) so this is never going to be used